lib/mongo_mapper/plugins/keys.rb

```diff
@@ -449,7 +449,6 @@ def internal_write_key(name, value, cast = true)
         else
           @_dynamic_attributes[key.name.to_sym] = as_typecast
         end
-        @attributes = nil
```

Comment:
Why does this need to be removed?

Reply:
Yes, indeed this is need to be removed. Both mongomapper and ActiveModel works around the structure of mixin (modules) layers, so this instance variable `@attributes` is expected to be owned by the actual model object.

But as far as I observed, internally, mongomapper doesn't actually make use this `@attributes` so removing here is kind of a no-op for rails 4.2.

But then in rails 5 onward, there're changes introduced by this [PR](https://github.com/rails/rails/pull/30985) which explicitly make use the `@attributes`, specifically in this line: https://github.com/rails/rails/pull/30985/commits/c3675f50d2e59b7fc173d7b332860c4b1a24a726#diff-dff03aefd62d372ccf26cfb9a350a0b2R67

With this change, removing this line become necessary as it causing our main attributes/columns storage to be cleared each time we update/set a field.